### DISCUSSION TITLE
Convert existing JSON to Pkl

### DISCUSSION THREAD

**Discussion by FranciscoSerrano** [apple/pkl#110] [Ideas]
I have lots of existing configs for projects, some are pretty long so I would love a quick way to convert from JSON to Pkl.

**Comment by kallekoo** [comment]
This would be an excellent feature. Dhall provides `json-to-dhall`, which not only enables converting existing configurations, but also editing structures via other means than a text editor, i.e. run `dhall-to-json`, make changes to JSON key paths, run `json-to-dhall`.

**Comment by StefMa** [comment]
There exist a json-schema to Pkl converter.
See https://github.com/apple/pkl-pantry/tree/main/packages/org.json_schema
[Here are the docs.](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/org.json_schema.contrib/current/generate/index.html) and this how to use it:
```
pkl eval package://pkg.pkl-lang.org/pkl-pantry/org.json_schema.contrib@<version>#/Generate.pkl \
  -m . \
  -p source="https://json.schemastore.org/github-action.json"
```

Its not json to Pkl, but at least something... 🤷

**Comment by bioball** [comment]
At the moment, converting JSON to Pkl requires writing a one-off converter. For an example of a converter, take a look at [k8s.contrib.convert](https://github.com/apple/pkl-pantry/blob/main/packages/k8s.contrib/convert.pkl).

Generally, a converter will parse JSON into Pkl, and then use the Pcf renderer as a poor man's Pkl renderer, with some extra logic to add imports, or amends headers.

This is something that we want to improve, though. It should be possible to provide a generic command for this.
